### PR TITLE
PW-7161 show discounted giftcard amount

### PR DESCRIPTION
**Pull request by zaiddreakh**
## Summary

refactoring of remaining amount container that shows up after applying a giftcard

**Fixed issue**:

### CODE REVIEW

File: src/cartridges/int_adyen_SFRA/cartridge/controllers/middlewares/adyen/partialPayment.js [hunk 1]
```diff
@@ -36,6 +36,12 @@ function makePartialPayment(req, res, next) {
       response.order.remainingAmount.currency,
     ).divide(100);
     response.remainingAmountFormatted = remainingAmount.toFormattedString();
+
+    const discountAmount = new Money(
+      response.amount.value,
+      response.amount.currency,
+    ).divide(100);
```
Comment (wboereboom) [inline src/cartridges/int_adyen_SFRA/cartridge/controllers/middlewares/adyen/partialPayment.js:43]: does this calculation check out for any currency? not just the ones using cents?

File: src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/checkoutConfiguration.js [hunk 2]
```diff
@@ -127,47 +128,50 @@ function removeGiftCard() {
...
-  const remainingAmountStart = document.createElement('div');
-  const remainingAmountEnd = document.createElement('div');
-  const cancelGiftCard = document.createElement('div');
-  const remainingAmountStartP = document.createElement('p');
-  const remainingAmountEndP = document.createElement('p');
-  const cancelGiftCardP = document.createElement('p');
-  const remainingAmountStartSpan = document.createElement('span');
-  const cancelGiftCardSpan = document.createElement('span');
-  const remainingAmountEndSpan = document.createElement('span');
+  Object.values(GIFTCARD_CONSTANTS).forEach((element) => {
```
Comment (maassenbas) [inline src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/checkoutConfiguration.js:133]: I'm not sure if it is a good practice to add a loop for each constant. If we suddenly would add another constant that is related to giftcards, then another container will be created here. 
I would suggest to create a function instead that will create these containers and then add in the parameters which type of container to create.

File: src/cartridges/int_adyen_SFRA/cartridge/client/default/js/constants.js [hunk 3]
```diff
@@ -4,4 +4,9 @@ module.exports = {
   METHOD_ADYEN: 'Adyen',
   METHOD_ADYEN_POS: 'AdyenPOS',
   METHOD_ADYEN_COMPONENT: 'AdyenComponent',
+  GIFTCARD_CONSTANTS: {
+    GIFTCARDAMOUNT: 'Giftcard Amount',
+    REMAININGAMOUNT: 'Remaining Amount',
+    CANCELGIFTCARD: 'Cancel Giftcard?',
```
Comment (maassenbas) [inline src/cartridges/int_adyen_SFRA/cartridge/client/default/js/constants.js:10]: Should we use Resource files instead to also enable translations on the storefront?

Comment (zaiddreakh) [reply src/cartridges/int_adyen_SFRA/cartridge/client/default/js/constants.js:10]: good point

File: src/cartridges/int_adyen_SFRA/cartridge/controllers/middlewares/adyen/partialPayment.js (hunk 1 above)
Comment (zaiddreakh) [reply src/cartridges/int_adyen_SFRA/cartridge/controllers/middlewares/adyen/partialPayment.js:43]: do we already have a way to handle all currencies? or do we need to create something new
I tested with USD, Euro, and Indian rubees and they work fine
Japenese Yen however doesn't need this division

Comment (wboereboom) [reply src/cartridges/int_adyen_SFRA/cartridge/controllers/middlewares/adyen/partialPayment.js:43]: You could have a look at the `getCurrencyValueForApi` method in AdyenHelper

**Pull request merged**